### PR TITLE
[MODULAR] OPFOR Expansion - Odds and Ends

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
This PR adds and cleans up some things in the OPFOR equipment list.
It adds:

- Pepperball gun
- Surplus Crate
- Super Surplus Crate
- Flashbang
It changes:

- Moves esword arm implant to implant section
- Clarifies the names of the AI multitool, chameleon projector, and chameleon no-slips
It removes:

- Non-chameleon thermal goggles for being redundant, we have the chameleon version anyway.

Also: a unit test! The unit test ensures that all OPFOR items (that aren't a parent) have some kind of `item_type` attached.
In addition it allows a per-equipment cap on how many can be ordered at once, all services are set to 1.
## How This Contributes To The Skyrat Roleplay Experience
More OPFOR stuff is fun, and it fixes things that bothered me.

## Changelog

:cl:
qol: More items in the OPFOR buylist, and some OPFOR item names have been clarified
fix: Spells now appear in the OPFOR list.
code: OFPOR item_type unit test added
/:cl:

### CODE REVIEW

File: code/modules/unit_tests/_unit_tests.dm [hunk 1]
```diff
@@ -115,7 +118,7 @@
 #include "timer_sanity.dm"
 #include "traitor.dm"
 #include "unit_test.dm"
-#include "wizard.dm"
+//#include "wizard.dm"
```
Comment (Gandalf2k15) [inline code/modules/unit_tests/_unit_tests.dm]: ?

File: code/modules/unit_tests/~skyrat/opposing_force_names.dm [hunk 2]
```diff
@@ -0,0 +1,8 @@
+///Checks that all OPFOR items have a name, either on the datum itself or the item it's attached to
```
Comment (Gandalf2k15) [inline code/modules/unit_tests/~skyrat/opposing_force_names.dm]: Put these all in one file

File: modular_skyrat/modules/opposing_force/code/equipment/equip_parent.dm [hunk 3]
```diff
@@ -1,3 +1,22 @@
+/// Global list of equipment parents, used for unit testing. PLEASE PLEASE PLEASE UPDATE THIS WITH NEW CATEGORIES AS THEY GET ADDED
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/opposing_force/code/equipment/equip_parent.dm]: I do not like having hard coded lists like this. People shouldn't have to add it he4e too tbh.

File: code/modules/unit_tests/_unit_tests.dm (hunk 1 above)
Comment (Zonespace27) [reply code/modules/unit_tests/_unit_tests.dm]: Accidentally left in, will fix

File: modular_skyrat/modules/opposing_force/code/equipment/equip_parent.dm (hunk 3 above)
Comment (Zonespace27) [reply modular_skyrat/modules/opposing_force/code/equipment/equip_parent.dm]: This list was the product of me trying to find a not-expensive method of getting only the direct subtypes of /datum/opposing_force_equipment, will give it another shot later

File: code/modules/unit_tests/_unit_tests.dm (hunk 1 above)
Comment (Zonespace27) [inline code/modules/unit_tests/_unit_tests.dm]:
```suggestion
#include "wizard.dm"
```

File: modular_skyrat/modules/opposing_force/code/equipment/equip_parent.dm (hunk 3 above)
Comment (Tastyfish) [reply modular_skyrat/modules/opposing_force/code/equipment/equip_parent.dm]: You only need to generate the list once during server initialization (probably during SSopposing_force startup). `subtypesof()` isn't _that_ expensive.

Comment (Zonespace27) [reply modular_skyrat/modules/opposing_force/code/equipment/equip_parent.dm]: Found a better way to do this!

File: code/modules/unit_tests/~skyrat/opposing_force_names.dm (hunk 2 above)
Comment (Zonespace27) [reply code/modules/unit_tests/~skyrat/opposing_force_names.dm]: Done

**Comment by Aether217** [comment]
Big no to the surplus crate. 

One big and important part about OPFORs is that we, as staff, manually approve all gear as to make sure things don't get out of hand. The crates are the opposite of that as they contain random items that we have no control over, a scenario that we do not want to be in as we push for as little vagueness as possible in them.

**Comment by softcerv** [comment]
> Big no to the surplus crate.
> 
> One big and important part about OPFORs is that we, as staff, manually approve all gear as to make sure things don't get out of hand. The crates are the opposite of that as they contain random items that we have no control over, a scenario that we do not want to be in as we push for as little vagueness as possible in them.

I think having these could be really cool for black market trader type gimmicks though. That being said, maybe a version of the crate could be made that has a much more slimmed down pool of items (or have the option to spawn it in with a set item selection.)

File: code/modules/unit_tests/_unit_tests.dm [hunk 4]
```diff
@@ -43,6 +43,7 @@
 /// A trait source when adding traits through unit tests
 #define TRAIT_SOURCE_UNIT_TESTS "unit_tests"
 
+#include "~skyrat/opposing_force.dm"
```
Comment (Gandalf2k15) [inline code/modules/unit_tests/_unit_tests.dm:47]: Skyrat comment

Comment (Zonespace27) [inline code/modules/unit_tests/_unit_tests.dm:47]:
```suggestion
// SKYRAT EDIT START
#include "~skyrat/opposing_force.dm"
//SKYRAT EDIT END
```

**Pull request merged**